safe/impact_function/impact_function.py

```diff
@@ -1884,7 +1884,7 @@ def pre_process(self):
         for pre_processor in self._preprocessors:
             layer = pre_processor['process']['function'](self)
             purpose = pre_processor['output'].get('value')['key']
-            result, name = self.datastore.add_layer(layer, purpose)
+            result, name = self.datastore.add_layer(layer, purpose, True)
```

Comment:
I'm not sure if I understand it correctly.
So actually the style was copied in temp folder but it doesn't get copied into the datastore?

Reply:
yes indeed. And it wouldn't be a good idea to save the style by default because we would have a QML file even if there is no style defined (not different than the default style provided by QGIS).
The datastore for now was not taking care about a possible previous style.